### PR TITLE
[Gen4] Added held item in wild generator/searcher

### PR DESCRIPTION
**Pull request by Real96**
Add what requested in #183 (gen4 only).
Thanks to @ShinySylveon04 for testing it!

### CODE REVIEW

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp [hunk 1]
```diff
@@ -366,6 +366,12 @@ std::vector<WildState> WildSearcher4::normalMethodJ(WildState state, u32 seed) c
     std::vector<WildState> states;
     state.setLead(Lead::None);
 
+    PokeRNG rng2(seed);
+    rng2.advance(4);
+    u8 heldItem = rng2.nextUShort() % 100;
+
+    state.setHeldItem(heldItem);
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Searchers/WildSearcher4.cpp:374]: Move to parent scope. Basically line 159 right before the if/else.

Will reduce code duplication and the amount of times this gets called.

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp [hunk 2]
```diff
@@ -394,6 +400,12 @@ std::vector<WildState> WildSearcher4::synchMethodJ(WildState state, u32 seed) co
     std::vector<WildState> states;
     state.setLead(Lead::Synchronize);
 
+    PokeRNG rng2(seed);
+    rng2.advance(4);
+    u8 heldItem = rng2.nextUShort() % 100;
+
+    state.setHeldItem(heldItem);
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Searchers/WildSearcher4.cpp:408]: Move to parent scope. Basically line 159 right before the if/else.

Will reduce code duplication and the amount of times this gets called.

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp [hunk 3]
```diff
@@ -428,6 +440,12 @@ std::vector<WildState> WildSearcher4::cuteCharmMethodJ(WildState state, u32 seed
 {
     std::vector<WildState> states;
 
+    PokeRNG rng2(seed);
+    rng2.advance(4);
+    u8 heldItem = rng2.nextUShort() % 100;
+
+    state.setHeldItem(heldItem);
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Searchers/WildSearcher4.cpp:448]: Move to parent scope. Basically line 159 right before the if/else.

Will reduce code duplication and the amount of times this gets called.

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp [hunk 4]
```diff
@@ -523,6 +541,12 @@ std::vector<WildState> WildSearcher4::normalMethodK(WildState state, u32 seed) c
...
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Searchers/WildSearcher4.cpp:549]: Move to parent scope. Basically line 237right before the if/else.

Will reduce code duplication and the amount of times this gets called.

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp [hunk 5]
```diff
@@ -552,6 +576,12 @@ std::vector<WildState> WildSearcher4::synchMethodK(WildState state, u32 seed) co
     std::vector<WildState> states;
     state.setLead(Lead::Synchronize);
 
+    PokeRNG rng2(seed);
+    rng2.advance(4);
+    u8 heldItem = rng2.nextUShort() % 100;
+
+    state.setHeldItem(heldItem);
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Searchers/WildSearcher4.cpp:584]: Move to parent scope. Basically line 237right before the if/else.

Will reduce code duplication and the amount of times this gets called.

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp [hunk 6]
```diff
@@ -586,6 +616,12 @@ std::vector<WildState> WildSearcher4::cuteCharmMethodK(WildState state, u32 seed
 {
     std::vector<WildState> states;
 
+    PokeRNG rng2(seed);
+    rng2.advance(4);
+    u8 heldItem = rng2.nextUShort() % 100;
+
+    state.setHeldItem(heldItem);
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Searchers/WildSearcher4.cpp:624]: Move to parent scope. Basically line 237right before the if/else.

Will reduce code duplication and the amount of times this gets called.

**Review by Admiral-Fish — CHANGES_REQUESTED** [review]
Please change instancea of "Held Item" to just "Item". Variable names or strings

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp (hunk 1 above)
Comment (Real96) [reply Source/Core/Gen4/Searchers/WildSearcher4.cpp]: Done

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp (hunk 2 above)
Comment (Real96) [reply Source/Core/Gen4/Searchers/WildSearcher4.cpp]: Done

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp (hunk 3 above)
Comment (Real96) [reply Source/Core/Gen4/Searchers/WildSearcher4.cpp]: Done

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp (hunk 4 above)
Comment (Real96) [reply Source/Core/Gen4/Searchers/WildSearcher4.cpp]: Done

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp (hunk 5 above)
Comment (Real96) [reply Source/Core/Gen4/Searchers/WildSearcher4.cpp]: Done

File: Source/Core/Gen4/Searchers/WildSearcher4.cpp (hunk 6 above)
Comment (Real96) [reply Source/Core/Gen4/Searchers/WildSearcher4.cpp]: Done

**Comment by Real96** [comment]
> Please change instancea of "Held Item" to just "Item". Variable names or strings

Done

**Pull request merged**